### PR TITLE
[ARRDT] Add pagination to example

### PR DESCRIPTION
**Pull request by alloy**
Makes use of `usePaginationFragment` to paginate the todo list.

### CODE REVIEW

**Comment by alloy** [comment]
Actually, I'm going to make one more addition, which is to show a loading state. As that's a typical need, even if synthetic in this example.

File: packages/apollo-react-relay-duct-tape/src/storeObservation/compiledHooks.ts [hunk 1]
```diff
@@ -412,7 +413,7 @@ function useLoadMore({
         disposable.current = undefined;
       }
     },
-    [disposable.current]
+    [] // On unmount
```
Comment (alloy) [inline packages/apollo-react-relay-duct-tape/src/storeObservation/compiledHooks.ts:416]: Actually quite an important bug-fix.

File: packages/apollo-react-relay-duct-tape/src/storeObservation/compiledHooks.test.tsx [hunk 2]
```diff
@@ -906,10 +906,14 @@ describe("compiledHooks", () => {
...
         beforeEach(async () => {
           onCompleted = jest.fn();
-          await act(() => {
+          await act(async () => {
             const { loadNext } = last(forwardUsePaginationFragmentResult);
             loadNext(1, { onCompleted });
 
+            // Introduce a slight delay before resolving the request as a
+            // regression test with a pagination requst being disposed early.
+            await new Promise((resolve) => setTimeout(resolve, 0));
```
Comment (alloy) [inline packages/apollo-react-relay-duct-tape/src/storeObservation/compiledHooks.test.tsx:915]: This is the regression test for that bug.

File: examples/apollo-watch-fragments/src/TodoList.tsx [hunk 3]
```diff
@@ -1,42 +1,66 @@
...
+  query: queryRef,
 }) => {
-  const todos = useFragment(TodoList_todosFragment, todosRef);
-  console.log("TodoList watch data:", todos);
+  const {
+    data: query,
+    hasNext,
+    loadNext,
+    isLoadingNext,
+  } = usePaginationFragment(
```
Comment (alloy) [inline examples/apollo-watch-fragments/src/TodoList.tsx:21]: Actual pagination example.

**Review by freiksenet — APPROVED** [review]

**Pull request merged**